packages/@lwc/features/src/__tests__/flags.spec.ts

```diff
@@ -0,0 +1,307 @@
+
+            if (runtimeFlags.ENABLE_FEATURE_NULL) {
+              if (ENABLE_FEATURE_TRUE) {
+                console.log('this looks like a bad idea');
+              }
+            }
+        `,
+    },
+};
+
+pluginTester({
+    title: 'non-prod environments',
+    plugin,
+    pluginOptions: {
+        featureFlags: {
+            ENABLE_FEATURE_TRUE: true,
+            ENABLE_FEATURE_FALSE: false,
+            ENABLE_FEATURE_NULL: null,
+        },
+    },
+    tests: nonProdTests,
+});
+
+pluginTester({
+    title: 'prod environments',
+    plugin,
+    pluginOptions: {
+        featureFlags: {
+            ENABLE_FEATURE_TRUE: true,
+            ENABLE_FEATURE_FALSE: false,
+            ENABLE_FEATURE_NULL: null,
+        },
+        prod: true,
+    },
+    tests: Object.assign({}, nonProdTests, {
+        // Override of nonProdTest version
+        'should transform boolean-true feature flags': {
+            code: `
+                import { ENABLE_FEATURE_TRUE } from '@lwc/features';
+                if (ENABLE_FEATURE_TRUE) {
+                    console.log('ENABLE_FEATURE_TRUE');
+                }
+            `,
+            output: `
+                import { ENABLE_FEATURE_TRUE, runtimeFlags } from '@lwc/features';
+                {
+                  console.log('ENABLE_FEATURE_TRUE');
+                }
+            `,
+        },
+        // Override of nonProdTest version
+        'should transform boolean-false feature flags': {
+            code: `
+                import { ENABLE_FEATURE_FALSE } from '@lwc/features';
+                if (ENABLE_FEATURE_FALSE) {
+                    console.log('ENABLE_FEATURE_FALSE');
+                }
+            `,
+            output: `
+                import { ENABLE_FEATURE_FALSE, runtimeFlags } from '@lwc/features';
+            `,
+        },
+        'should transform both boolean and null feature flags': {
+            code: `
+                import { ENABLE_FEATURE_TRUE, ENABLE_FEATURE_FALSE, ENABLE_FEATURE_NULL } from '@lwc/features';
+                if (ENABLE_FEATURE_TRUE) {
+                    console.log('ENABLE_FEATURE_TRUE');
+                }
+                if (ENABLE_FEATURE_FALSE) {
+                    console.log('ENABLE_FEATURE_FALSE');
+                }
+                if (ENABLE_FEATURE_NULL) {
+                    console.log('ENABLE_FEATURE_NULL');
+                }
+            `,
+            output: `
+                import { ENABLE_FEATURE_TRUE, ENABLE_FEATURE_FALSE, ENABLE_FEATURE_NULL, runtimeFlags } from '@lwc/features';
+                {
+                  console.log('ENABLE_FEATURE_TRUE');
+                }
+
+                if (runtimeFlags.ENABLE_FEATURE_NULL) {
+                  console.log('ENABLE_FEATURE_NULL');
+                }
+            `,
+        },
+        'should transform runtime flag lookups into compile-time flags': {
+            code: `
+                if (runtimeFlags.ENABLE_FEATURE_TRUE) {
```

Comment:
I don't understand this code... why is this using `runtimeFlags`?

Reply:
This is to align with our build process where the dist files that are generated by rollup in `@lwc/engine` are processed again by rollup in a subsequent build for production.